shell/server/identity.js

```diff
@@ -57,9 +58,72 @@ makeIdentity = (identityId, requriements) => {
 
 globalFrontendRefRegistry.register({
   frontendRefField: "identity",
+  typeId: Identity.typeId,
 
   restore(db, saveTemplate, identityId) {
     return new Capnp.Capability(new IdentityImpl(db, saveTemplate, identityId),
                                 IdentityRpc.PersistentIdentity);
   },
+
+  validate(db, session, identityId, options) {
```

Comment:
I've updated the signature of `validate()`. Now it takes in `frontendRefRequest`, which must be type-checked, and returns `frontendRef`.